### PR TITLE
Make map size configurable

### PR DESCRIPTION
**Pull request by jayanthvn**
*Issue #, if available:* N/A

*Description of changes:*
If caller has to modify any map data, they could use the new API. Right now only map size is configurable and it is the responsibility of the caller to reload the BPF map

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/elfparser/elf.go [hunk 1]
```diff
@@ -310,6 +356,7 @@ func (e *elfLoader) parseSection() error {
 	for index, section := range e.elfFile.Sections {
 		if section.Name == "license" {
 			data, err := section.Data()
+			log.Infof("JAY - %s", string(data))
```
Comment (achevuru) [inline pkg/elfparser/elf.go:344]: stale log line

File: pkg/elfparser/elf.go [hunk 2]
```diff
@@ -138,7 +161,30 @@ func (b *bpfSDKClient) LoadBpfFile(path, customizedPinPath string) (map[string]B
...
+		return nil, nil, err
+	}
+	return bpfLoadedProg, bpfLoadedMaps, nil
+}
+
+func (b *bpfSDKClient) LoadBpfFileWithCustomData(inputData BpfCustomData) (map[string]BpfData, map[string]ebpf_maps.BpfMap, error) {
+
+	bpfFile, err := os.Open(inputData.FilePath)
+	if err != nil {
+		log.Infof("LoadBpfFile failed to open")
```
Comment (achevuru) [inline pkg/elfparser/elf.go:160]: Should be an error instead right? Also fn name in the log line doesn't match (or) maybe we can just say - file create error or something along those lines

File: pkg/elfparser/elf.go [hunk 3]
```diff
@@ -408,6 +455,13 @@ func (e *elfLoader) parseMap() ([]ebpf_maps.CreateEBPFMapInput, error) {
 			}
 		}
 		log.Infof("Found map name %s", mapData.Name)
+
+		if len(customData.CustomMapSize) != 0 {
```
Comment (achevuru) [inline pkg/elfparser/elf.go:444]: I think, it might be safe to validate the size value passed in. Cap it @ 1M with maybe the lower bound being @ 64k

Comment (jayanthvn) [reply pkg/elfparser/elf.go:443]: Oh this is just checking if custom Map is present

Comment (achevuru) [reply pkg/elfparser/elf.go:443]: I mean, once we identify we've a non Zero value in `CustomMapSize`, I'm proposing we cap min and max values before we set the `mapData.MaxDntries` field below. We can probably cap the configurable value in the client as well (i.e.,) NP agent ..

Comment (jayanthvn) [reply pkg/elfparser/elf.go:443]: Yeah I think the caller can check since this is SDK

**Review by achevuru — APPROVED** [review]

**Pull request merged**